lib/puppet/resource_api/transport_schema_def.rb

```diff
@@ -0,0 +1,8 @@
+require 'puppet/resource_api/base_type_definition'
+# Provides accessor methods for the type being provided
+class Puppet::ResourceApi::TransportSchemaDef < Puppet::ResourceApi::BaseTypeDefinition
+  def initialize(definition)
+    raise Puppet::DevError, 'TransportSchemaDef requires definition to be a Hash' unless definition.is_a?(Hash)
+    super(definition, :connection_info)
+  end
+end
```

Comment:
Given how small `TransportSchemaDef` and `TypeDefinition` have become I'd pref to have them all in one file:

```
module Puppet::ResourceApi
  # pre-declare class
  class BaseTypeDefinition; end
 
  class Puppet::ResourceApi::TypeDefinition < Puppet::ResourceApi::BaseTypeDefinition
    # ...
  ...
  ..
  class BaseTypeDefinition
    # the rest
```